Source/DafnyStandardLibraries/src/Std/TargetSpecific/FileIO5-notarget-java-cs-js-go-py.dfy

```diff
@@ -47,4 +52,113 @@ module Std.FileIO {
     var isError, errorMsg := FileIOInternalExterns.INTERNAL_WriteBytesToFile(path, bytes);
     return if isError then Failure(errorMsg) else Success(());
   }
+
+  /**
+    * Attempts to read a text file and convert its contents to a string.
+    * The file must contain valid UTF-8 encoded text.
+    *
+    * Parameters:
+    * - fileName: string - The path to the file to read
+    *
+    * Returns:
+    * - Result<string, string> - On success, returns the file contents as a string.
+    *   On failure, returns an error message describing what went wrong.
+    *
+    * Error cases:
+    * - File does not exist or cannot be accessed
+    * - File content is not valid UTF-8
+    * - File contains characters that cannot be represented as UTF-16
+    *
+    * Example:
+    * ```dafny
+    * var result := FileIO.ReadUTF8FromFile("example.txt");
+    * if result.IsFailure() {
+    *   print "Error reading file: ", result.error;
+    * } else {
+    *   print "File contents: ", result.value;
+    * }
+    * ```
+    */
+  method ReadUTF8FromFile(fileName: string) returns (r: Result<string, string>) {
+    var bytes :- ReadBytesFromFile(fileName);
+    return UnicodeStringsWithUnicodeChar.FromUTF8Checked(seq(|bytes|, i requires 0 <= i < |bytes| => bytes[i] as uint8));
```

Comment:
SO much better, thank you! The rest of the declarations below aside from `WriteUTF8ToFile` are dead code and can be deleted now, right?